### ISSUE TITLE
Feature Request: add BlockIgnores and TokenIgnores settings, similar to Vale

### ISSUE THREAD

**Issue by landure** [rvben/rumdl#798]
### Problem or motivation

I'm using [Hugo](https://gohugo.io/) as a CMS, and use its shortcode feature.

One of my shortcode is named `nodejs`, and it raise issues with `MD044` rule:

```
[MD044] Proper name 'nodejs' should be 'Node.js'
```

The shortcode is called between moustaches in markdown:

```md
{{% nodejs %}}
```

`rumdl` doesn't allow to exclude Hugo shortcodes from checks.

[Vale](https://vale.sh/) solve this with `BlockIgnores` and `TokenIgnores` rules:

```ini
# IgnorePatterns for Hugo static site generator.
BlockIgnores = (?s)({{[%<].*?[%>]}})
# TokenIgnore for Hugo static site generator short-codes, path and URLs.
TokenIgnores = (?s) *(\[[^\]]*\]\([^)]*\)), \
(?s)({{[^}]*}}), \
(?s)^({#{[^}]+})
```

### Proposed solution

Implement `block_ignores` and `token_ignores` global, and per rules settings.

### Alternatives considered

I could use inline rules, but ignoring CMS template inclusions is a good feature to have.

### Feature category

CLI / configuration

### Example markdown

```markdown
{{% nodejs %}}
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Thanks for the report and the reproduction. Two answers, because there are two
things in it.

The bug is fixed and released. Since v0.2.52 a Hugo shortcode invocation
(`{{% ... %}}` and `{{< ... >}}`) is opaque to
MD033, MD037, MD038, MD044, MD045, MD057 and MD085, so `{{% nodejs %}}` is
neither reported by MD044 nor rewritten to `{{% Node.js %}}` by `rumdl fmt`.
Your `hugo.md` now reports only line 3, the prose, and the zero-config `fmt`
corruptions of ``{{% x `code ` %}}`` and `{{% x * y * %}}` are gone. The body
between paired shortcode tags is still linted; for the `{{% %}}` notation Hugo
renders that body as Markdown, and for `{{< >}}` it is up to the shortcode's
template, which rumdl cannot see, so linting it is the safe default.

The general `BlockIgnores` / `TokenIgnores` setting I am not going to add. The
single Vale name hides two mechanisms with very different costs here: a
"suppress findings inside these ranges" filter is cheap but does not make MD013
stop counting the characters or MD022 stop seeing the blank lines around a
construct, and a "treat these spans as opaque" mode is per-rule work that
cannot be done centrally, because a regex cannot lie to a CommonMark parser.
Vale gets away with the simpler model because it is a prose linter over
tokens. A permanent global regex-ignore surface next to inline
`rumdl-disable`, per-file ignores and flavors would be a third way to say the
same thing, half of which cannot work.

What I would take instead, if you or others hit it: rumdl already recognizes
Jinja and Liquid `{{ ... }}` / `{% ... %}` ranges, but MD037, MD038 and MD044
do not yet consult them the way they consult shortcodes, so a Jekyll or
MkDocs-macros document still has this exposure. If you have a document where
one of those bites, open it with the input and I will give those rules the
same guard.

**Comment by landure** [comment]
Thank you for the feedback.

**Comment by rvben** [comment]
Thanks for the proposal. We’re closing this because a global regex-ignore surface would add a second matching language with broad, hard-to-predict suppression behavior. The supported direction remains structural scopes and targeted inline/config suppression. The durable rationale is documented in [the out-of-scope decision](https://github.com/rvben/rumdl/blob/main/.out-of-scope/regex-ignore-surfaces.md).
